src/DependencyInjection/Compiler/ExtensionCompilerPass.php

```diff
@@ -215,6 +235,30 @@ protected function hasTrait(\ReflectionClass $class, $traitName)
         return $this->hasTrait($parentClass, $traitName);
     }
 
+    /**
```

Comment:
```suggestion
    /**
     * @phpstan-param 'excludes'|'admins'|'implements'|'extends'|'instanceof'|'uses' $type
```

Reply:
phpstan can't understand this:

```
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  Line   DependencyInjection/Compiler/ExtensionCompilerPass.php
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  121    Parameter #1 $type of method Sonata\AdminBundle\DependencyInjection\Compiler\ExtensionCompilerPass::isSubtypeOf() expects 'admins'|'excludes'|'extends'|'implements'|'instanceof'|'uses', string given.
  219    Method Sonata\AdminBundle\DependencyInjection\Compiler\ExtensionCompilerPass::flattenExtensionConfiguration() should return array<'admins'|'excludes'|'extends'|'implements'|'instanceof'|'uses',
         array<string, array<string, array<string, array('priority' => string)>>>> but returns array<string, array<string, array<string, array('priority' => array<string, string>)>>>.
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

```

Reply:
It's because `type` is considered as string, because of 
```
@param array<string, array<string, array<string, array<string, array<string, mixed>>>>> $extensionMap
```

Maybe it's better to keep string, since `isSubtypeOf` does support every possible string ; there is a default `return false`.